torch/fx/graph.py

```diff
@@ -137,6 +137,22 @@ def graph_copy(self, g : 'Graph', val_map : Dict[Node, Node]) -> Optional[Argume
             val_map[node] = self.node_copy(node, lambda n : val_map[n])
         return None
 
+    def __deepcopy__(self, memo=None) -> 'Graph':
+        """
+        Explicitly implement __deepcopy__ to prevent excessive recursion depth
+        from the default implementation. This uses graph_copy to copy the nodes
+        in an interative way, rather than recursive. It also populates the
+        memoization table to prevent unnecessary copies (e.g. references to
+        nodes or other parts of the Graph from a custom GraphModule implementation
+        """
+        g = Graph()
+        output_val = g.graph_copy(self, val_map=memo)
+        g.output(output_val)
+        hashable_attrs = ['_root', '_insert', '_len']
+        for attr in hashable_attrs:
+            memo[getattr(self, attr)] = getattr(g, attr)
```

Comment:
I don't really understand what this part is doing. If we are taking over the `__deepcopy__` implementation for `Graph`, we don't need to populate this memo table, right? Who would be consulting it?

Reply:
The idea was that there could be references to these attributes from outside the Graph, but considering these attributes are all private I think we can just drop it